src/sentry/region_to_control/producer.py

```diff
@@ -1,91 +1,178 @@
+from __future__ import annotations
```

Comment:
All of this is from https://github.com/getsentry/sentry/pull/40353/files so feel free to ignore